client/src/js/dd_main.js

```diff
@@ -40,9 +40,15 @@
     var xTimeBuffer = 1000;
     var motor_power = new streamGraph.StreamingDataset(motor_data, { color: 'rgb(88, 86, 214)' });
     motor_power.filter(function (data) {
-        return data.filter(function (datum) {
-            return datum.x >= timeDomain.begin() - xTimeBuffer;
-        });
+        var time_window = timeDomain.begin() - xTimeBuffer;
```

Comment:
Should this be moved into `StreamingDataset`? Seems like a common use-case.

Reply:
It already is part of that? I think you are looking at the JS not the TS. The JS is all autogenerated.

Reply:
Oh, that's confusing. Can we add the generated files to .gitignore?

Reply:
We didn't want to have to install and configure typescript on the raspberry pi. It is already a hassle to build the go binary hence why we tried to upload everything to here.

Reply:
We can't use github releases like codegen-tooling?

Reply:
We can but that's a lot of infra to auto-download on the PI side and I don't want to spend time on before Sunday. Happy to work on this in the coming weeks.

Reply:
SGTM